constraint/pkg/client/drivers/local/driver.go

```diff
@@ -287,25 +296,45 @@ func (d *Driver) Query(ctx context.Context, target string, constraints []*unstru
 
 		kindResults, err := drivers.ToResults(constraintsMap, resultSet)
 		if err != nil {
-			return nil, nil, err
-		}
-
-		for _, result := range kindResults {
-			result.EvaluationMeta = RegoEvaluationMeta{
-				TemplateRunTime: float64(evalEndTime.Nanoseconds()) / 1000000,
-				ConstraintCount: uint(len(kindResults)),
-			}
+			return &drivers.QueryResponse{Err: err}
 		}
 
 		results = append(results, kindResults...)
+
+		if d.gatherMetrics {
+			// for the batch of evaluated constraints, report their evaluation metadata
+			evalRunTime := float64(evalEndTime.Nanoseconds()) / 1000000
+
+			statsEntries = append(statsEntries,
+				&instrumentation.StatsEntry{
+					Key: kind,
+					Stats: []*instrumentation.Stat{
+						{
+							Name:        templateRunTimeMillisName,
```

Comment:
For multi-engine, knowing what engine created these metrics could also be useful

Reply:
+1 to that! We are capturing the engine type lower here on line 318